### PR TITLE
New Workflow: Initial commit for Generate_Register_files.

### PR DESCRIPTION
**Pull request by sihyung-maxim**
New "Generate_Register_Files" Workflow. Similar to /clang-format-run, the user will just have to update the peripheral SVD files and run the following command to automatically generate and push in the register files and <part_number>.svd file for them. It's less work for the PR owner.

Command:
`/generate-register-files [VALID_PART_NUMBER]`

Where [VALID_PART_NUMBER] could be **me18**, **max32690**, **ME14**, **MAX78000**, etc. Capitalization doesn't matter and you can use either the die name or chip name. **Note:** [VALID_PART_NUMBER] is an optional argument. You can just run
`/generate-register-files` to generate the register files for all parts dependent on your peripheral SVD file change. **For most users,** `/generate-register-files` **is the recommended command to use.**

If you want to generate the register files for all parts, then you can run **all** for [VALID_PART_NUMBER], but I wouldn't recommend using this parameter unless for extreme cases such as updating the copyright notices in all the parts. 
`/generate-register-files all`

The bot will also acknowledge your commented command by reacting with a 'rocket' emoji when it begins processing. If the bot was successfully able to generate and push in the register files and <part_number>.svd file, then the bot will react with the 'hooray' emoji. If there was an issue, then the bot will react with a 'confused' emoji and will comment there's an issue. For specific issues, please check the Actions log.

**Note:** Because this is the initial commit for this new workflow, the bot will not work until it's pushed into the main branch. I will create a new PR demoing this bot after this PR is merged.

Use-case Examples:
**1)** Say you updated gcr_me18.svd. The following command will trigger the bot to generate and push in the updated _gcr_regs.h_ and _max32690.svd_ files.

`/generate-register-files me18`

**2)** Say you updated _icc_reva.svd_, a generic peripheral SVD file which used for several parts. Instead of running `/generate-register-files part1`, `/generate-register-files part2`, ... `/generate-register-files partn`, multiple times to update all _icc_regs.h_, you can just run:

`/generate-register-files`

### CODE REVIEW

**Comment by sihyung-maxim** [comment]
Like the Verify_Register_SVD workflow, this will also have to run on the btm-ci machine because of the private repo authentication checkout issues.

**Review by Jacob-Scheiffler — COMMENTED** [review]
Seems legit. I don't have any experience writing workflow scripts so I'll leave the full code review to those that actually know what they're looking at. I do have a thought/question/comment about the functionality though. Is it possible to have the script generate the register files without the chip specified? Like, if someone just commented "/generate-register-files" could you determine which SVD files changed and then update the associated register files?

**Comment by sihyung-maxim** [comment]
> Seems legit. I don't have any experience writing workflow scripts so I'll leave the full code review to those that actually know what they're looking at. I do have a thought/question/comment about the functionality though. Is it possible to have the script generate the register files without the chip specified? Like, if someone just commented "/generate-register-files" could you determine which SVD files changed and then update the associated register files?

That's a good idea. Didn't even think of that when writing this current version... That's way better because then the PR owner won't have to worry about generating register files for multiple parts if they edited a generic peripheral SVD file linked to multiple parts. I'll create another job in this same workflow, so the user has the option to specify the part or not.

**Review by Jake-Carter — APPROVED** [review]

**Review by kevin-gillespie — APPROVED** [review]

File: .github/workflows/Generate_Register_Files.yml [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+        run: |
+          git clone git@github.com:Analog-Devices-MSDK/msdk-internal.git
+
+      - name: Generating register files.
+        run: |
+          # Create associative array for dictionary
+          declare -A CHIP_TO_DIE_NAMES 
+          declare -A DIE_TO_CHIP_NAMES
+
+          # Add future parts to this dictionary
```
[the remark this answers is not in the document]
Comment (sihyung-maxim) [reply .github/workflows/Generate_Register_Files.yml:79]: In the context of bash, do you mean declaring a single associative array housing all conversions (die -> chip and chip -> die names)?

**Comment by sihyung-maxim** [comment]
> > Seems legit. I don't have any experience writing workflow scripts so I'll leave the full code review to those that actually know what they're looking at. I do have a thought/question/comment about the functionality though. Is it possible to have the script generate the register files without the chip specified? Like, if someone just commented "/generate-register-files" could you determine which SVD files changed and then update the associated register files?
> 
> That's a good idea. Didn't even think of that when writing this current version... That's way better because then the PR owner won't have to worry about generating register files for multiple parts if they edited a generic peripheral SVD file linked to multiple parts. I'll create another job in this same workflow, so the user has the option to specify the part or not.

Just added your suggested feature @Jacob-Scheiffler.

**Comment by sihyung-maxim** [comment]
Of course, this is the initial commit for the new workflow. I'll start testing this in a new PR once this PR is merged into main.

**Review by Jacob-Scheiffler — APPROVED** [review]

**Pull request merged**